### PR TITLE
Features/people details

### PR DESCRIPTION
**Pull request by patrycja-gomza**
👋 Hello Team!
I'm excited to present this pull request, which brings several enhancements and additions to our project:

1. People Details Saga and Slice:
   - Added saga and slice to handle people details fetching.
   - Implemented API call to retrieve details for a person.

2. PersonDetails Component:
   - Created the `PersonDetails` component to display details for a specific person.
   - Utilizes the new saga and slice for fetching and managing data.

3. People Details Tile Component:
   - Added the `PersonDetailsTile` component for presenting person details.
   - Displays poster, title, overview, date of birth, and place of birth.

4. Root Saga Update:
   - Extended the root saga to include the new `watchFetchPersonDetails` saga.

5. Redux Store Update:
   - Expanded the Redux store to include the `peopleDetails` slice.

6. Search Movie URL Update:
   - Updated the search movie URL in the APIURLS file.

7. People List Component Update:
   - Enhanced the `PeopleList` component to include clickable links to individual people details.

8. APIURLS Modification:
   - Added a new temporary API URL for person's details.

📝 Additional Notes:
- Please note that my code is not functioning correctly yet🚨
- I am currently facing an issue with accessing the `details` property for a person 😢😢
- I would appreciate the reviewer's assistance in identifying and resolving the problem 👀

Feel free to provide any feedback or suggestions for improvement.

Thank you! 🙌

### CODE REVIEW

**Comment by bedicooper** [comment]
Thank You, awsome work!

I will have an opportunity to copy Your PR in the evening at the earliest. Could You elaborate on the: 

> issue with accessing the `details` property for a person

I don't see `details` prop in the object we get from API.

Regarding the Tiles, as we agreed I'm currently working on a generic Tile for Movie and Person details, that works just like what you wrote (only uses more generic prop names). I should be done with it today or by Jan 10.
I think the best course of action is for You to merge this PR when aproved and then open a new one to edit this section and use new common Tiles, what do You think?

**Comment by bedicooper** [comment]
OK, it looks like React is trying to render components and asks store for data but the store is still empty.
So You ask for instance for `poster={details.profile_path}` but empty object from initialState is _undefined_ and has no _profile_path_ property, hence the error.

Unfortunately, I won't be able today to test your code and find the solution as I hoped 😢

File: src/features/people/PeopleDetails/index.js [hunk 1]
```diff
@@ -1,7 +1,45 @@
...
+                            <PersonDetailsTile
+                                poster={details.profile_path}
+                                title={details.name}
+                                overview={details.biography}
+                                dateOfBirth={details.birthday}
+                                placeOfBirth={details.place_of_birth}
+                            />
+                        </>
+                    }
+                    />
```
Comment (bedicooper) [inline src/features/people/PeopleDetails/index.js:38]: lines 27, 37 and 38 will break Your render, remove them if You want <PersonDetailsTile> to render :)

File: src/features/people/PeopleDetails/peopleDetailsSlice.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const peopleDetailsSlice = createSlice({
+    name: "personDetails",
```
Comment (bedicooper) [inline src/features/people/PeopleDetails/peopleDetailsSlice.js:4]: Hej! I found your mistake :)
to Redux the slice od state You are creating is called peopleDetails but then You put diferent value in name property.
So your Selector was asking for non existing slice 😄 

change that line to 
`name: "peopleDetails",`

you also have to correct it in line 30
💪

File: src/features/people/PeopleDetails/peopleDetailsSlice.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        },
+    },
+});
+export const {
+    fetchDataSuccess,
+    getDetailsForPerson,
+    fetchDataError,
+} = peopleDetailsSlice.actions;
+
+export const selectPersonState = (state) => state.personDetails;
```
Comment (bedicooper) [inline src/features/people/PeopleDetails/peopleDetailsSlice.js:30]: `const selectPersonState = (state) => state.peopleDetails;
`
PS: you don't need to export that selector as it is only used in Slice

File: src/features/people/PeopleDetails/peopleDetailsSaga.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    fetchDataSuccess,
+    selectPersonId,
+    fetchDataError,
+    getDetailsForPerson,
+} from "./peopleDetailsSlice";
+import { getPeopleDetails } from "./getPeopleDetails";
+
+function* fetchPersonDetailsHandler() {
+    try {
+        console.log("Saga: Handling getDetailsForPerson action");
```
Comment (bedicooper) [inline src/features/people/PeopleDetails/peopleDetailsSaga.js:12]: Do we need that log in deployed app?

File: src/features/people/PeopleDetails/index.js [hunk 5]
```diff
@@ -1,7 +1,45 @@
...
+} from "./peopleDetailsSlice";
+import { PersonDetailsTile } from "./PersonDetailsTile";
+
+const PersonDetails = () => {
+    const dispatch = useDispatch();
+    const { id } = useParams();
+
+    useEffect(() => {
+        dispatch(getDetailsForPerson({ personId: id }));
+        console.log("Dispatching getDetailsForPerson action");
```
Comment (bedicooper) [inline src/features/people/PeopleDetails/index.js:17]: Do we need that log in deployed app?

**Review by bedicooper — COMMENTED** [review]
Hey, great job, thank you!

I found two mistakes, if You correct those lines Your code runs perfectly 💪 

I appreciate that, from the beginning, you handle an error by rendering a simple message as we didn't implement an Error Page yet.

File: src/features/people/PeopleDetails/peopleDetailsSaga.js (hunk 4 above)
Comment (patrycja-gomza) [reply src/features/people/PeopleDetails/peopleDetailsSaga.js]: Thank you for catching that! 😀The console log was only for debugging purposes during the coding phase, and I forgot to remove it. It has no purpose in the deployed app, so I have now removed the unnecessary logs 😁
Appreciate your sharp eye for details! 😀

File: src/features/people/PeopleDetails/index.js (hunk 5 above)
Comment (patrycja-gomza) [reply src/features/people/PeopleDetails/index.js]: Thank you for catching that! 😀The console log was only for debugging purposes during the coding phase, and I forgot to remove it. It has no purpose in the deployed app, so I have now removed the unnecessary logs 😁
Appreciate your sharp eye for details! 😀

File: src/features/people/PeopleDetails/index.js (hunk 1 above)
Comment (patrycja-gomza) [reply src/features/people/PeopleDetails/index.js]: A big thanks for your valuable assistance! 🙌 Your sharp eye caught those two mistakes, and I've made the necessary corrections. Your guidance was instrumental, and I've learned a lot through this process. 🌟
I appreciate your time and support. The code is now running seamlessly! 🚀

File: src/features/people/PeopleDetails/peopleDetailsSlice.js (hunk 2 above)
Comment (patrycja-gomza) [reply src/features/people/PeopleDetails/peopleDetailsSlice.js]: See above for the detailed response. Thanks again! 😀🙏🌟

File: src/features/people/PeopleDetails/peopleDetailsSlice.js (hunk 3 above)
Comment (patrycja-gomza) [reply src/features/people/PeopleDetails/peopleDetailsSlice.js]: Thanks a lot for your observation and suggestion! 🙏 I appreciate your input. I've made the adjustment and removed the unnecessary export for `selectPersonState`.
Feel free to share more insights if you have them! 😊

**Comment by patrycja-gomza** [comment]
## Pull Request Update: Code Fixes 🛠️

Hi team! 🌟

I've corrected all the issues, taking into account the valuable feedback from  @bedicooper . The code is now free of errors and ready for review 🚀 

I would be grateful if you could take a look at the changes and approve them. Thanks in advance! 🙏

**Review by bedicooper — APPROVED** [review]
Send it!
🚀🚀🚀

**Comment by widmo200** [comment]
Hi ! everything is fine, but I don't see the photo of the actor I choose

**Comment by patrycja-gomza** [comment]
> Hi ! everything is fine, but I don't see the photo of the actor I choose

Hi! 👋Thanks for checking it out! I appreciate your feedback. I'll make sure to address the missing photo of the actor in the next step. Your input is valuable 🙌😊

**Pull request merged**